Microsoft.HealthVault/ItemTypes/ApproximateDateTime.cs

```diff
@@ -542,6 +542,11 @@ public int CompareTo(DateTime other)
         ///
         public override bool Equals(object obj)
         {
+            if (obj?.GetType() != this.GetType())
```

Comment:
CompareTo supports DateTime objects in addition to ApproximateDateTime objects.  It might be more appropriate to try/catch and return false in the catch, or at least check specifically for DateTimes and let them pass through to CompareTo.